### PR TITLE
Fixes issue #4892.

### PR DESCRIPTION
**Pull request by logan-dev-oss**
First contribution here 😊 ! Do not hesitate to correct me. 

This PR is related to issue #4892 . 

# Summary 

```rust
-literal.method_call(args)
```
The main idea is to not trigger `clippy::precedence` when the method call is an odd function.

# Example 

```rust
// should trigger lint 
let _ = -1.0_f64.abs() //precedence of method call abs() and neg ('-') is ambiguous

// should not trigger lint
let _ = -1.0_f64.sin() // sin is an odd function => -sin(x) = sin(-x) 
```

# Theory

Rust allows following literals: 
- char
- string
- integers
- floats
- byte
- bool

Only integers/floats implements the relevant `std::ops::Neg`.
Following odd functions are implemented on i[8-128] and/or f[32-64]: 
- `asin`
- `asinh`
- `atan`
- `atanh`
- `cbrt`
- `fract`
- `round`
- `signum`
- `sin`
- `sinh`
- `tan`
- `tanh `
- `to_degrees`
- `to_radians`

# Implementation 

As suggested by `flip1995` in [comment](https://github.com/rust-lang/rust-clippy/issues/4892#issuecomment-568249683), this PR add a whitelist of odd functions and compare method call to the the whitelist before triggering lint. 

changelog: Don't trigger [`clippy::precedence`] on odd functions.

### CODE REVIEW

File: clippy_lints/src/precedence.rs [hunk 1]
```diff
@@ -86,11 +103,16 @@ impl EarlyLintPass for Precedence {
...
+            if let ExprKind::MethodCall(ref path_segment, ref args) = rhs.kind {
                 if let Some(slf) = args.first() {
                     if let ExprKind::Lit(ref lit) = slf.kind {
                         match lit.kind {
                             LitKind::Int(..) | LitKind::Float(..) => {
+                                for &odd_function in &ODD_FUNCTIONS_WHITELIST {
+                                    if odd_function == &*path_segment.ident.name.as_str() {
+                                        return;
+                                    }
+                                }
```
Comment (matthiaskrgr) [inline clippy_lints/src/precedence.rs:115]: We could use an iterator here for more idiomatic rust:

````rust
if ODD_FUNCTIONS_WHITELIST
    .iter()
    .any(|odd_function| **odd_function == *path_segment.ident.name.as_str())
{
    return;
};
````

Comment (matthiaskrgr) [inline clippy_lints/src/precedence.rs:115]: I order to not compute `&*path_segment.ident.name.as_str()` for every `odd_function`, we can cache the result in a variable outside of the loop and reuse it:

````rust
....
           let path_seg_str = path_segment.ident.name.as_str(); // this is new
                if let Some(slf) = args.first() {
                    if let ExprKind::Lit(ref lit) = slf.kind {
                        match lit.kind {
                            LitKind::Int(..) | LitKind::Float(..) => {
                                if ODD_FUNCTIONS_WHITELIST
                                    .iter()
                                    .any(|odd_function| *odd_function == &*path_seg_str) // use here
                                {
                                    return;
                                };
````

Comment (logan-dev-oss) [reply clippy_lints/src/precedence.rs]: Integrated

**Comment by logan-dev-oss** [comment]
I integrated recommended changes and rebase upstream/master.

**Comment by flip1995** [comment]
@bors r+

Thanks!

**Pull request merged**